modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2363,17 +2363,21 @@ def explode(self, column):
     # UDF (apply and agg) methods
     # There is a wide range of behaviors that are supported, so a lot of the
     # logic can get a bit convoluted.
-    def apply(self, func, axis, *args, **kwargs):
+    def apply(self, func, axis, is_series, *args, **kwargs):
         # if any of args contain modin object, we should
         # convert it to pandas
+
+        # TODO(vkarthik): currently is_series only works for callable
+        # functions that return Series objects. We should handle this
+        # for other cases as well.
         args = try_cast_to_pandas(args)
         kwargs = try_cast_to_pandas(kwargs)
         if isinstance(func, dict):
             return self._dict_func(func, axis, *args, **kwargs)
         elif is_list_like(func):
             return self._list_like_func(func, axis, *args, **kwargs)
         else:
-            return self._callable_func(func, axis, *args, **kwargs)
+            return self._callable_func(func, axis, is_series, *args, **kwargs)
```

Comment:
I'm not sure if this is the best place to put this logic. Does anyone know where else we could put this information?